crates/pathfinder/src/state/sync.rs

```diff
@@ -734,13 +736,31 @@ fn update_starknet_state(
         // Update the global state tree.
         storage_commitment_tree
             .set(contract_address, contract_state_hash)
-            .context("Updating global state tree")?;
+            .context("Updating storage commitment tree")?;
     }
 
-    // Apply all global tree changes.
-    storage_commitment_tree
+    // Apply storage commitment tree changes.
+    let new_storage_commitment = storage_commitment_tree
+        .apply()
+        .context("Apply storage commitment tree updates")?;
+
+    let class_commitment_tree = ClassCommitmentTree::load(transaction, class_commitment)
+        .context("Loading class commitment tree")?;
+
+    // FIXME 0.11.0 once the gateway's StateUpdate gets the declared_classes field
+    #[cfg(fixme_0_11_0)]
```

Comment:
Is this `cfg` something you're trying out?

Reply:
`fixme_0_11_0` is not defined anywhere so this is just an alternative way of commenting out code as I didn't have access to your updated gateway types